### ISSUE TITLE
NPM crash after updating package Centos 7 - Epel Repo

### ISSUE THREAD

**Issue by ZRothy1** [npm/npm#17589]
#### I'm opening this issue because:

  - [X] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [X] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?

Good Afternoon,

I recently had updated Node and NPM as I was working on a Cordova project which suggested or required updating NodeJS. After I updated both packages, NPM crashes with errors similar to `failed to fetch from registry: https://registry.npmjs.org/normalize-package-data` or `failed to fetch from registry: https://registry.npmjs.org/cordova`.

It then outputs a debug log which I don't really get as I don't have too much experience with NPM and Node.

```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node',
1 verbose cli   '/usr/bin/npm',
1 verbose cli   'install',
1 verbose cli   '-g',
1 verbose cli   'cordova@6.5.0' ]
2 info using npm@3.10.10
3 info using node@v6.10.3
4 silly loadCurrentTree Starting
5 silly install loadCurrentTree
6 silly install readGlobalPackageData
7 silly fetchPackageMetaData cordova@6.5.0
8 silly fetchNamedPackageData cordova
9 silly mapToRegistry name cordova
10 silly mapToRegistry using default registry
11 silly mapToRegistry registry https://registry.npmjs.org/
12 silly mapToRegistry data Result {
12 silly mapToRegistry   raw: 'cordova',
12 silly mapToRegistry   scope: null,
12 silly mapToRegistry   escapedName: 'cordova',
12 silly mapToRegistry   name: 'cordova',
12 silly mapToRegistry   rawSpec: '',
12 silly mapToRegistry   spec: 'latest',
12 silly mapToRegistry   type: 'tag' }
13 silly mapToRegistry uri https://registry.npmjs.org/cordova
14 verbose request uri https://registry.npmjs.org/cordova
15 verbose request no auth needed
16 info attempt registry request try #1 at 4:52:19 PM
17 verbose request id 3c780193ec88d934
18 http request GET https://registry.npmjs.org/cordova
19 http 200 https://registry.npmjs.org/cordova
20 verbose headers { 'nginx/1.10.1application/jsonsun, 02 jul 2017 12:31:31 gmtw/"5958e7a3-16d564"gzipmax-age=30052504bytesmon, 03 jul 2017 20:52:19 gmt1.1 varnish5keep-alivecache-ewr18132-ewrhit1s1499115140.532298,vs0,ve0accept-encoding': '' }
21 silly get cb [ 200,
21 silly get   { 'nginx/1.10.1application/jsonsun, 02 jul 2017 12:31:31 gmtw/"5958e7a3-16d564"gzipmax-age=30052504bytesmon, 03 jul 2017 20:52:19 gmt1.1 varnish5keep-alivecache-ewr18132-ewrhit1s1499115140.532298,vs0,ve0accept-encoding': '' } ]
22 silly fetchPackageMetaData Error: failed to fetch from registry: https://registry.npmjs.org/cordova
22 silly fetchPackageMetaData     at /usr/lib/node_modules/npm/lib/cache/caching-client.js:164:27
22 silly fetchPackageMetaData     at /usr/lib/node_modules/npm/lib/cache/caching-client.js:53:8
22 silly fetchPackageMetaData     at f (/usr/lib/node_modules/npm/node_modules.bundled/once/once.js:25:25)
22 silly fetchPackageMetaData     at /usr/lib/node_modules/npm/node_modules.bundled/npm-registry-client/lib/request.js:76:16
22 silly fetchPackageMetaData     at /usr/lib/node_modules/npm/node_modules.bundled/npm-registry-client/lib/request.js:103:12
22 silly fetchPackageMetaData     at f (/usr/lib/node_modules/npm/node_modules.bundled/once/once.js:25:25)
22 silly fetchPackageMetaData     at CachingRegistryClient.<anonymous> (/usr/lib/node_modules/npm/node_modules.bundled/npm-registry-client/lib/request.js:297:12)
22 silly fetchPackageMetaData     at Request._callback (/usr/lib/node_modules/npm/node_modules.bundled/npm-registry-client/lib/request.js:210:14)
22 silly fetchPackageMetaData     at Request.self.callback (/usr/lib/node_modules/npm/node_modules.bundled/request/request.js:187:22)
22 silly fetchPackageMetaData     at emitTwo (events.js:106:13)
22 silly fetchPackageMetaData  error for cordova@6.5.0 Error: failed to fetch from registry: https://registry.npmjs.org/cordova
22 silly fetchPackageMetaData     at /usr/lib/node_modules/npm/lib/cache/caching-client.js:164:27
22 silly fetchPackageMetaData     at /usr/lib/node_modules/npm/lib/cache/caching-client.js:53:8
22 silly fetchPackageMetaData     at f (/usr/lib/node_modules/npm/node_modules.bundled/once/once.js:25:25)
22 silly fetchPackageMetaData     at /usr/lib/node_modules/npm/node_modules.bundled/npm-registry-client/lib/request.js:76:16
22 silly fetchPackageMetaData     at /usr/lib/node_modules/npm/node_modules.bundled/npm-registry-client/lib/request.js:103:12
22 silly fetchPackageMetaData     at f (/usr/lib/node_modules/npm/node_modules.bundled/once/once.js:25:25)
22 silly fetchPackageMetaData     at CachingRegistryClient.<anonymous> (/usr/lib/node_modules/npm/node_modules.bundled/npm-registry-client/lib/request.js:297:12)
22 silly fetchPackageMetaData     at Request._callback (/usr/lib/node_modules/npm/node_modules.bundled/npm-registry-client/lib/request.js:210:14)
22 silly fetchPackageMetaData     at Request.self.callback (/usr/lib/node_modules/npm/node_modules.bundled/request/request.js:187:22)
22 silly fetchPackageMetaData     at emitTwo (events.js:106:13)
23 silly rollbackFailedOptional Starting
24 silly rollbackFailedOptional Finishing
25 silly runTopLevelLifecycles Finishing
26 silly install printInstalled
27 verbose stack Error: failed to fetch from registry: https://registry.npmjs.org/cordova
27 verbose stack     at /usr/lib/node_modules/npm/lib/cache/caching-client.js:164:27
27 verbose stack     at /usr/lib/node_modules/npm/lib/cache/caching-client.js:53:8
27 verbose stack     at f (/usr/lib/node_modules/npm/node_modules.bundled/once/once.js:25:25)
27 verbose stack     at /usr/lib/node_modules/npm/node_modules.bundled/npm-registry-client/lib/request.js:76:16
27 verbose stack     at /usr/lib/node_modules/npm/node_modules.bundled/npm-registry-client/lib/request.js:103:12
27 verbose stack     at f (/usr/lib/node_modules/npm/node_modules.bundled/once/once.js:25:25)
27 verbose stack     at CachingRegistryClient.<anonymous> (/usr/lib/node_modules/npm/node_modules.bundled/npm-registry-client/lib/request.js:297:12)
27 verbose stack     at Request._callback (/usr/lib/node_modules/npm/node_modules.bundled/npm-registry-client/lib/request.js:210:14)
27 verbose stack     at Request.self.callback (/usr/lib/node_modules/npm/node_modules.bundled/request/request.js:187:22)
27 verbose stack     at emitTwo (events.js:106:13)
28 verbose cwd /home/zach/hosted/mobileapp
29 error Linux 3.10.0-123.13.2.el7.x86_64
30 error argv "/usr/bin/node" "/usr/bin/npm" "install" "-g" "cordova@6.5.0"
31 error node v6.10.3
32 error npm  v3.10.10
33 error failed to fetch from registry: https://registry.npmjs.org/cordova
34 error If you need help, you may report this error at:
34 error     <https://github.com/npm/npm/issues>
35 verbose exit [ 1, true ]
```

Thanks in advance for the help in figurign this out.

#### How can the CLI team reproduce the problem?

I updated NodeJS and NPM via yum on my Centos 7 install from the epel-release repository. Afterwards things such as npm install or npm install -g cordova@6.5.0 crashes. My epel repository is listed as the following url: 

```
[epel]
name=Extra Packages for Enterprise Linux 7 - $basearch
baseurl=http://download.fedoraproject.org/pub/epel/7/$basearch
#mirrorlist=https://mirrors.fedoraproject.org/metalink?repo=epel-7&arch=$basearch
failovermethod=priority
enabled=1
gpgcheck=1
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-7

[epel-debuginfo]
name=Extra Packages for Enterprise Linux 7 - $basearch - Debug
#baseurl=http://download.fedoraproject.org/pub/epel/7/$basearch/debug
mirrorlist=https://mirrors.fedoraproject.org/metalink?repo=epel-debug-7&arch=$basearch
failovermethod=priority
enabled=0
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-7
gpgcheck=1

[epel-source]
name=Extra Packages for Enterprise Linux 7 - $basearch - Source
#baseurl=http://download.fedoraproject.org/pub/epel/7/SRPMS
mirrorlist=https://mirrors.fedoraproject.org/metalink?repo=epel-source-7&arch=$basearch
failovermethod=priority
enabled=0
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-7
gpgcheck=1
```

Current Node / NPM package installed are as follows:
```
nodejs.x86_64                             1:6.10.3-1.el7             @epel      
nodejs-devel.x86_64                       1:6.10.3-1.el7             @epel      
nodejs-packaging.noarch                   7-1.el7                    @epel
npm.x86_64                                1:3.10.10-1.6.10.3.1.el7   @epel
```

### supporting information:

 - `npm -v` prints: 3.10.10
 - `node -v` prints: v6.10.3
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: Linux, Centos 7
 - Network issues:
   - Geographic location where npm was run:
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by jimbo8098** [comment]
Hi KenanY,

If I'm not mistaken, it looks like the issue in the debug log is network related. Are you able to ping https://registry.npmjs.org/cordova ? Try a wget on the file too, maybe ping is blocked/simply returned from the firewall.

If it's fine, were any updates made to RHEL's kernel? I know they introduced SELinux on Cent 7 and that caused all manner of unexpected issues when I was supporting it. Such issues should be reported in the system logs or (more likely) audit logs.

Hope this helps!

-Jim

**Comment by ZRothy1** [comment]
I was able to wget, curl as well as ping without any issue. As for updates I don't normally update much on the machine having this trouble as it is my work PC. It was recommended not to update the OS on it so it doesn't break the dev environment at all. It was only recently that after I updated the install of npm that I was unable to install anything with npm via cordova commands or npm itself. Prior to the updating the packages, I was able to download/install cordova plugins without issue. 

While in the past I have encountered issues where my office firewall was causing issues downloading from repositories, it usually blocked ping as well. 

```
ping registry.npmjs.org
PING a.sni.fastly.net (151.101.208.162) 56(84) bytes of data.
64 bytes from 151.101.208.162: icmp_seq=1 ttl=62 time=3.34 ms
64 bytes from 151.101.208.162: icmp_seq=2 ttl=62 time=3.21 ms
64 bytes from 151.101.208.162: icmp_seq=3 ttl=62 time=3.27 ms
64 bytes from 151.101.208.162: icmp_seq=4 ttl=62 time=3.23 ms
```

**Comment by legodude17** [comment]
Does this occur with the latest version of `npm`?

**Comment by ZRothy1** [comment]
I got around to testing the last LTS package from node's website rather than the latest as its version was closest to what the epel repository had. That LTS package did not have the issue. The NPM version of the package was 3.10.10, which was the same as the npm I installed fromt he LTS package. Only the node version differed, with node being 6.10.3, and the LTS package having node 6.11.1.

So seems it was just something with the RPM I installed from epel when updating node. 

Thanks for the help.
